doc/mgmt/SONiC_ConfigDB_support_Yang_default_values_and_profiles.md

```diff
@@ -0,0 +1,392 @@
+  - [2.6 Code example](#26-code-example)
+- [3 Reboot](#3-reboot)
+  - [3.1 Cold-reboot](#31-cold-reboot)
+  - [3.2 Warm-reboot](#31-warm-reboot)
+  - [3.3 Fast-reboot](#31-fast-reboot)
+  - [3.4 Schema upgrade and DB migration](#34-schema-upgrade-and-db-migration)
+- [4 Error handling](#4-error-handling)
+- [5 Serviceability and Debug](#5-serviceability-and-debug)
+- [6 Unit Test](#6-unit-test)
+- [7 Migration steps](#7-migration-steps)
+  - [7.1 Phase 1](#71-phase-1)
+  - [7.2 Phase 2](#72-phase-2)
+- [8 References](#8-references)
+  - [SONiC YANG MODEL GUIDELINES](#sonic-yang-model-guidelines)
+
+# About this Manual
+
+This document provides a detailed description on the new features for:
+
+- Yang model default value.
+- Profile DB.
+- swss-common API change.
+
+## Terminologies
+
+- Yang model:
+  
+  - The Yang model define a hierarchical data structure.
+  - SONiC Define config DB schema with Yang model, please refer to [SONiC YANG MODEL GUIDELINES](#7-1-sonic-yang-model-guidelines)
+
+- Default value:
+  
+  - Yang model support define default value for configuration items.
+  - For example, default value for nat_zone:
+    
+    ```
+        leaf nat_zone {
+            description "NAT Zone for the vlan interface";
+            type uint8 {
+                range "0..3" {
+                    error-message "Invalid nat zone for the vlan interface.";
+                    error-app-tag nat-zone-invalid;
+                }
+            }
+                                default "0";
+        }
+    ```
+
+- J2 template:
+  
+  - SONiC using Jinja2 template generate configuration files during deply minigraph.
+  
+  - For example, buffer config are rendered by J2 template:
+    
+    ```
+        {%- set default_cable = '40m' %}
+        {%- macro generate_buffer_pool_and_profiles() %}
+            "BUFFER_POOL": {
+                "ingress_lossless_pool": {
+                    "size": "26531072",
+                    "type": "ingress",
+                    "mode": "dynamic",
+                    "xoff": "6291456"
+                },
+                "egress_lossless_pool": {
+                    "size": "32822528",
+                    "type": "egress",
+                    "mode": "static"
+                }
+            },
+    
+            ......
+    
+        {%- endmacro %}
+    ```
+
+- Profile
+  
+  - Profile is part of OS image, should upgrade with OS upgrade.
+  - Profile are suggested values based on expirence.
+  - User can overwrite profile.
+
+## Problem Statement
+
+- SONiC still using old default value and config from j2 template after OS upgrade:
+  - Following config may update after OS upgrade:
+    - Default value: defined in yang model.
+    - Profile: defined in j2 template.
+  - Currently all config stored in config DB, so above configs can't be update after OS upgrade.
+- Potential risk, Yang model default value conflict with hardcoded value:
+  - Default value hardcoded in source code.
```

Comment:
Just to double check, did you mean the hardcoded value in the templates?

Reply:
No, it's ins the source code, for example in hostcfgd, when read config from config DB, if some item does not exist in config DB, hostcfgd will use default values hardcoded in it's code.